src/Domain/FunctionsOfTime/ReadSpecPiecewisePolynomial.cpp

```diff
@@ -31,6 +31,7 @@ void read_spec_piecewise_polynomial(
     const bool quaternion_rotation) {
   h5::H5File<h5::AccessType::ReadOnly> file{file_name};
   for (const auto& [spec_name, spectre_name] : dataset_name_map) {
+    file.close_current_object();
```

Comment:
That isn't the best place. Just below this on line 45-46, there's a `continue`. If the closing was at the end of the loop, it'd also have to be there just before the `continue`, otherwise a file would still be open on the next iteration of the loop (since it didn't make it to the bottom). Having it at the top just before we open the file guarantees that no other objects are open and reduces the amount of places you have to remember to close the file.

Reply:
Okay, fair enough!